### PR TITLE
Moving the DatabaseLoader support into Microsoft.ML.Data

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/machinelearning/issues/4323

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @eerhardt, @CESARDELATORRE

**Comment by eerhardt** [comment]
I believe we can remove this file now:

https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Experimental/Properties/AssemblyInfo.cs

It was only added for the DatabaseLoader change.

Also, I believe we can revert remove this line as well:

https://github.com/dotnet/machinelearning/blob/461b7d38c06d64e53c76e29770665c4b0bb44e72/src/Microsoft.ML.Core/Properties/AssemblyInfo.cs#L25

Basically, we can undo the extra workarounds in #4035.

It looks like there are 2 workarounds that can be undone from #4091. (Adding 2 `[BestFriend]` attributes.)
